vault/logical_system.go

```diff
@@ -2515,6 +2516,66 @@ func (b *SystemBackend) handleMetrics(ctx context.Context, req *logical.Request,
 	return b.Core.metricsHelper.ResponseForFormat(format), nil
 }
 
+// Note that although this is a streaming response, it times out and the client connection is closed
+// after 90 seconds. I haven't been able to find the place where that's set, or override it.
+func (b *SystemBackend) handleMonitor(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	ll := data.Get("log_level").(string)
+	w := req.ResponseWriter
+	resp := &logical.Response{}
+
+	if ll == "" {
+		ll = "INFO"
+	}
+	logLevel := log.LevelFromString(ll)
+
+	if logLevel == log.NoLevel {
+		return logical.ErrorResponse("unknown log level"), nil
+	}
+
+	flusher, ok := w.ResponseWriter.(http.Flusher)
+	if !ok {
+		return logical.ErrorResponse("streaming not supported"), nil
+	}
+
+	isJson := b.Core.LogFormat() == "json"
+	logger := b.Core.Logger().(log.InterceptLogger)
+
+	mon, _ := monitor.NewMonitor(512, logger, &log.LoggerOptions{
+		Level:      logLevel,
+		JSONFormat: isJson,
+	})
+	defer mon.Stop()
+
+	logCh := mon.Start()
+	w.WriteHeader(http.StatusOK)
+
+	// 0 byte write is needed before the Flush call so that if we are using
+	// a gzip stream it will go ahead and write out the HTTP response header
+	_, err := w.Write([]byte(""))
+
+	if err != nil {
```

Comment:
It's good practice to wrap errors, e.g. `return resp, fmt.Errorf("error seeding flusher: %w", err)` or so. This is not just to be verbose, but specifically if there are various ways we might see stdlib error values, it helps to disambiguate if someone comes to us seeking support...by keeping these error values distinct we know exactly where it came from.

Reply:
Makes sense, will do